libcxx/test/std/iterators/iterator.primitives/range.iter.ops/range.iter.ops.advance/iterator_count_sentinel.pass.cpp

```diff
@@ -188,11 +229,12 @@ constexpr bool test() {
         check_forward_sized_sentinel<int*>(                        range, range+size, n, expected);
       }
 
-      {
-        // Note that we can only test ranges::advance with a negative n for iterators that
-        // are sized sentinels for themselves, because ranges::advance is UB otherwise.
-        // In particular, that excludes bidirectional_iterators since those are not sized sentinels.
+      // Exclude the `n == 0` case for the backwards checks.
```

Comment:
Question: why do we need to exclude `n == 0`?

Reply:
I guess it isn't strictly needed. I thought it would be more appropriate for the `check_backward` tests to only test the `n < 0` case as the `n == 0` case is covered by the `check_forward` tests above. I added a comment to that effect.

Reply:
Thanks for explaining! I'd remove the conditional and just check regardless -- IMO it's simpler, and having some overlap is okay.

Reply:
Yep, the `check_backwards` tests also work for the `n == 0` case, as expected.